lib/zip/entry_set.rb

```diff
@@ -14,9 +14,7 @@ def include?(entry)
     end
 
     def find_entry(entry, case_sensitively = true)
-      return @entry_set[to_key(entry)] if case_sensitively
-      entry = @entry_set.find { |k, _| k.downcase == to_key(entry).downcase }
-      entry.last if entry
+      @entry_set[to_key(entry)]
```

Comment:
I don't like it.
There must be a variant where `case_insensitive_match ` is `false`, but we use `case_sensitively ` as `false`